aspnetcore/tutorials/first-mvc-app-xplat/adding-controller.md

```diff
@@ -0,0 +1,102 @@
+
+![Browser window showing an application response of This is the Welcome action method](../first-mvc-app/adding-controller/_static/welcome.png)
+
+Let's modify the example slightly so that you can pass some parameter information  from the URL to the controller (for example, `/HelloWorld/Welcome?name=Scott&numtimes=4`).  Change the `Welcome` method  to include two parameters as shown below. Note that the code uses the C# optional-parameter feature to indicate that the `numTimes` parameter defaults to 1 if no value is passed for that parameter.
+
+[!code-csharp[Main](../first-mvc-app/start-mvc/sample/MvcMovie/Controllers/HelloWorldController.cs?name=snippet_2)]
+
+The code above uses `HtmlEncoder.Default.Encode` to protect the app from malicious input (namely JavaScript). It also uses [Interpolated Strings](https://docs.microsoft.com/dotnet/articles/csharp/language-reference/keywords/interpolated-strings).
+
+Run the app and browse to:
+
+   http://localhost:5000/HelloWorld/Welcome?name=Rick&numtimes=4
+
+You can try different values for `name` and `numtimes` in  the URL. The MVC [model binding](xref:mvc/models/model-binding) system automatically maps the named parameters from  the query string in the address bar to parameters in your method. See [Model Binding](xref:mvc/models/model-binding) for more information.
+
+![Browser window showing an application response of Hello Rick, NumTimes is: 4](../first-mvc-app/adding-controller/_static/rick4.png)
+
+In the sample above, the URL segment (`Parameters`) is not used, the `name` and `numTimes` parameters are passed as [query strings](http://en.wikipedia.org/wiki/Query_string). The `?` (question mark) in the above URL is a separator, and the query strings follow. The `&` character separates query strings.
+
+Replace the `Welcome` method with the following code:
+
+[!code-csharp[Main](../first-mvc-app/start-mvc/sample/MvcMovie/Controllers/HelloWorldController.cs?name=snippet_3)]
+
+Run the app and enter the following URL:  `http://localhost:xxx/HelloWorld/Welcome/3?name=Rick`
+
+![Browser window showing an application response of Hello Rick, ID: 3](../first-mvc-app/adding-controller/_static/rick_routedata.png)
+
+This time the third URL segment  matched the route parameter `id`. The `Welcome`  method contains a parameter  `id` that matched the URL template in the `MapRoute` method. The trailing `?`  (in `id?`) indicates the `id` parameter is optional.
+
+[!code-csharp[Main](../first-mvc-app/start-mvc/sample/MvcMovie/Startup.cs?name=snippet_1&highlight=5)]
+
+In these examples the controller has been doing the "VC" portion  of MVC - that is, the view and controller work. The controller is returning HTML  directly. Generally you don't want controllers returning HTML directly, since  that becomes very cumbersome to code and maintain. Instead we'll typically use a separate Razor view template file to help generate the HTML response. We'll do that in the next tutorial.
```

Comment:
If you're going to publish this without the next tutorial in the series, should probably remove the direction to go to the next tutorial